osu.Game/Online/OnlineViewContainer.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Game.Online.API;
+using osu.Game.Online.Placeholders;
+
+namespace osu.Game.Online
+{
+    /// <summary>
+    /// A <see cref="Container"/> for dislaying online content who require a local user to be logged in.
+    /// Shows its children only when the local user is logged in and supports displaying a placeholder if not.
+    /// </summary>
+    public class OnlineViewContainer : Container, IOnlineComponent
+    {
+        private readonly Container content;
+        private readonly Container placeholderContainer;
+        private readonly Placeholder placeholder;
+
+        private const int transform_time = 300;
+
+        protected override Container<Drawable> Content => content;
```

Comment:
This and the underlying field is a bit confusing. An improperly-written derived class could override this get-only protected property to return something different, but have to call the base ctor, in which case `content` is initialised in the private field in the base, but not accessible to the derived, therefore creating one dangling container reference per instance. You probably want a protected auto property here.